logstash-core/src/main/java/org/logstash/instrument/witness/schedule/ScheduledWitness.java

```diff
@@ -0,0 +1,44 @@
+package org.logstash.instrument.witness.schedule;
+
+import java.time.Duration;
+
+/**
+ * A witness that is self-populating on a given schedule.
+ */
+public interface ScheduledWitness {
+
+    /**
+     * The duration between updates for this witness
+     *
+     * @return the {@link Duration} between scheduled updates. For example {@link Duration#ofMinutes(long)} with a value of 5 would schedule this implemenation to
+     * self-populate every 5 minute. Defaults to every 60 seconds. - Note, implementations may not allow schedules faster then every 1 second.
+     */
+    default Duration every() {
+        //note - the system property is an only an escape hatch if this proves to cause performance issues. Do not document this system property, it is not part of the contract.
+        return Duration.ofSeconds(Long.parseLong(System.getProperty("witness.scheduled.duration.in.seconds", "10")));
+    }
+
+    /**
+     * Get the name to set for the thread on which this is scheduled. This is useful for debugging purposes. Defaults to the class name + -thread.
+     *
+     * @return The name for the scheduled thread.
+     */
+    default String threadName() {
+        return getClass().getSimpleName() + "-thread";
+    }
+
+    /**
+     * Sets the priority for the underlying thread. Consumers should NOT set this unless they are absolutely certain. Defaults to {@link Thread#NORM_PRIORITY}
+     *
+     * @return the thread scheduled thread priority
+     */
+    default int priority() {
+        //note - the system property is an only an escape hatch if this proves to cause performance issues. Do not document this system property, it is not part of the contract.
+        return Integer.parseInt(System.getProperty("witness.schedule.priority", Integer.toString(Thread.NORM_PRIORITY)));
```

Comment:
@jakelandis any reason why we'd want to expose this (even internally)?

Reply:
No specific reason.  My thought process was more of a general template to implement background services.  If it doesn't make sense, I can remove.

Reply:
@jakelandis At least I don't see a valid reason for manually setting this here ever => I'd kill it :)